File: src/main/java/tn/esprit/pitest11/Entities/Entretien.java
package tn.esprit.pitest11.Entities;


import tn.esprit.pitest11.Entities.User;

import java.time.LocalDate;
import java.util.Objects;

public class Entretien {

    private int id;
    private User user;
    private LocalDate date;
    private String Lieu;
    private double longitude;
    private double latitude;
    private Recrutement recrutement;
    private boolean approved;




    public Recrutement getRecrutement() {
        return recrutement;
    }

    public void setRecrutement(Recrutement recrutement) {
        this.recrutement = recrutement;
    }

    public Entretien(int id,User user, LocalDate date, String lieu, double longitude, double latitude, Recrutement recrutement, boolean approved) {
        this.id = id;
        this.user = user;
        this.date = date;
        Lieu = lieu;
        this.longitude = longitude;
        this.latitude = latitude;
        this.recrutement = recrutement;
        this.approved = approved;
    }

    public Entretien() {
    }

    public Entretien(int id, User user, LocalDate date, String lieu, Recrutement recrutement) {
        this.id = id;
        this.user = user;
        this.date = date;
        Lieu = lieu;
        this.recrutement = recrutement;
    }


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public String getLieu() {
        return Lieu;
    }

    public void setLieu(String lieu) {
        Lieu = lieu;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public boolean isApproved() {
        return approved;
    }

    public void setApproved(boolean approved) {
        this.approved = approved;
    }

    @Override
    public String toString() {
        return "Entretien{" +
                "id=" + id +
                ", user=" + user +
                ", date=" + date +
                ", Lieu='" + Lieu + '\'' +
                ", longitude=" + longitude +
                ", latitude=" + latitude +
                ", recrutement=" + recrutement +
                ", approved=" + approved +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Entretien entretien = (Entretien) o;
        return id == entretien.id && Objects.equals(date, entretien.date);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, date);
    }
}

File: src/Entity/Entretien.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity]
class Entretien
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: "integer")]
    private int $id;

    #[ORM\Column(type: "integer")]
    private int $userId;

    #[ORM\Column(type: "date")]
    #[Assert\NotNull(message: "La date de l'entretien est requise.")]
    private \DateTimeInterface $date;

    #[ORM\Column(type: "string", length: 255)]
    #[Assert\NotBlank(message: "Le lieu ne peut pas être vide.")]
    private string $lieu;

    #[ORM\Column(type: "float")]
    #[Assert\NotBlank(message: "Longitude is required.")]
    #[Assert\Type(type: "float", message: "Longitude must be a valid floating-point number.")]
    #[Assert\GreaterThan(value: -180, message: "Longitude must be greater than -180.")]
    #[Assert\LessThan(value: 180, message: "Longitude must be less than 180.")]
    private float $longitude;

    #[ORM\Column(type: "float")]
    #[Assert\NotBlank(message: "Latitude is required.")]
    #[Assert\Type(type: "float", message: "Latitude must be a valid floating-point number.")]
    #[Assert\GreaterThan(value: -90, message: "Latitude must be greater than -90.")]
    #[Assert\LessThan(value: 90, message: "Latitude must be less than 90.")]
    private float $latitude;

    #[ORM\Column(type: "integer")]
    private int $recrutementId;

    #[ORM\Column(type: "boolean")]
    private bool $approved;

    #[ORM\ManyToOne(targetEntity: Recrutement::class, inversedBy: 'entretiens')]
    #[ORM\JoinColumn(nullable: false)]
    private Recrutement $recrutement;

    public function getId(): int
    {
        return $this->id;
    }

    public function setId(int $value): void
    {
        $this->id = $value;
    }

    public function getUserId(): int
    {
        return $this->userId;
    }

    public function setUserId(int $value): void
    {
        $this->userId = $value;
    }

    public function getDate(): \DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $value): void
    {
        $this->date = $value;
    }

    public function getLieu(): string
    {
        return $this->lieu;
    }

    public function setLieu(string $value): void
    {
        $this->lieu = $value;
    }

    public function getLongitude(): float
    {
        return $this->longitude;
    }

    public function setLongitude(float $value): void
    {
        $this->longitude = $value;
    }

    public function getLatitude(): float
    {
        return $this->latitude;
    }

    public function setLatitude(float $value): void
    {
        $this->latitude = $value;
    }

    public function getRecrutementId(): int
    {
        return $this->recrutementId;
    }

    public function setRecrutementId(int $value): void
    {
        $this->recrutementId = $value;
    }

    public function getApproved(): bool
    {
        return $this->approved;
    }

    public function setApproved(bool $value): void
    {
        $this->approved = $value;
    }

    public function getRecrutement(): Recrutement
    {
        return $this->recrutement;
    }

    public function setRecrutement(Recrutement $recrutement): void
    {
        $this->recrutement = $recrutement;
    }
}
